### ISSUE TITLE
[BUG] training data is inadequate

### ISSUE THREAD

**Issue by ilya-rusyanov** [open-spaced-repetition/fsrs4anki#589]
**Describe the bug**

Today I've upgraded FSRS optimizer (python) to the latest version an ran it on decks in batch as usual. One of decks has failed with message "training data is inadequate".

**To Reproduce**
1. Run optimizer on deck.
2. Get 'training data is inadequate' error

**Expected behavior**
Optimizer runs correctly.

**Screenshots**
![unnamed](https://github.com/open-spaced-repetition/fsrs4anki/assets/1244545/99305214-0598-45d9-8bf8-f03c5bc68f43)

**Environment**
 - Anki version: 2.1.66
 - OS: Debian 12.4
 - Scheduler version: 4.11.1
 - Optimizer version: 4.21.0

**Additional context**
What is unusual about the decks is that a month ago

* I've changed the settings so that the review daily limit to 2 cards
* Almost every card's type was modified from a regular basic type to some custom.

I'm eager to share the deck but not publicly.

**Comment by L-M-Sherlock** [comment]
The log shows that there are only 128 reviews for training. It's not enough.

**Comment by ilya-rusyanov** [comment]
Oh, god. Have not noticed that. Sorry.
